matlab/Dockerfile

```diff
@@ -0,0 +1,153 @@
+# Copyright 2020 The MathWorks, Inc.
+
+# Argument shared across multi-stage build to hold location of installed MATLAB 
+ARG BASE_ML_INSTALL_LOC=/tmp/matlab-install-location
+
+# For finding latest versions of the base image see
+# https://github.com/SwissDataScienceCenter/renkulab-docker
+ARG RENKU_BASE_IMAGE=renku/renkulab-py:3.9-0.10.3
+
+# Replace "mathworks/matlab" with any Docker image that contains MATLAB
+# MATLAB should be available on the path in the Docker image
+FROM mathworks/matlab AS matlab-install-stage
+ARG BASE_ML_INSTALL_LOC
+
+# Run code to locate a MATLAB install in the base image and softlink
+# to BASE_ML_INSTALL_LOC for a latter stage to copy 
+RUN export ML_INSTALL_LOC=$(which matlab) \
+    && if [ ! -z "$ML_INSTALL_LOC" ]; then \
+        ML_INSTALL_LOC=$(dirname $(dirname $(readlink -f ${ML_INSTALL_LOC}))); \
+        echo "soft linking: " $ML_INSTALL_LOC " to" ${BASE_ML_INSTALL_LOC}; \
+        ln -s ${ML_INSTALL_LOC} ${BASE_ML_INSTALL_LOC}; \
+    elif [ $BASE_ML_INSTALL_LOC = '/tmp/matlab-install-location' ]; then \
+        echo "MATLAB was not found in your image."; exit 1; \
+    else \
+        echo "Proceeding with user provided path to MATLAB installation: ${BASE_ML_INSTALL_LOC}"; \
+    fi
+
+FROM ${RENKU_BASE_IMAGE}
+# Use Renku instead of the base notebook
+# FROM jupyter/base-notebook
+ARG BASE_ML_INSTALL_LOC
+
+# Switch to root user
+USER root
+
+# Copy MATLAB install from supplied Docker image
+COPY --from=matlab-install-stage ${BASE_ML_INSTALL_LOC} /usr/local/MATLAB
+
+# Put MATLAB on the PATH
+RUN ln -s /usr/local/MATLAB/bin/matlab /usr/local/bin/matlab
+
+# Install MATLAB dependencies
+# Reference: https://github.com/mathworks-ref-arch/container-images/tree/master/matlab-deps
+# The following are the dependencies for R2021a
+RUN export DEBIAN_FRONTEND=noninteractive && apt-get update && apt-get install --no-install-recommends -y \
```

Comment:
it's a lot of stuff here - should we make a matlab base image instead? If it's all in the template we don't actually have a good way of knowing if it even builds.